### DISCUSSION TITLE
How to validate selection?

### DISCUSSION THREAD

**Discussion by ilya2204** [ianstormtaylor/slate#5631] [Q&A]
Hello, I have a selection and I want to validate it. I want to know if I can safely set it in the editor without getting an error later. What's the best way to do this?

**Comment by gerardnico** [comment]
Can you give an example of selection with a validation that can give an error later?

I have a hard time trying to understand your validation.

**Reply by ilya2204** [reply]
For example this code gives error:
``` ts
    const point = Editor.end(editor, []);

    point.offset += 1;

    editor.select({ anchor: point, focus: point});
```

`Cannot resolve a DOM point from Slate point: {"path":[1,0,0],"offset":13}`

The end of the editor is not the only case. Similar cases may occur in the middle blocks of the editor

**Reply by gerardnico** [reply]
This error should not occur.

That's mostly because the DOMs (Slate React and Browser) are not in sync.

You may wait for the end of the rendering with a `useEffect` when working with location (ranges, ...)

Something like that:
```jsx
useEffect(() => {
        
        editor.select({ anchor: point, focus: point});
        
}, [point]);
```
